guides.Rmd

```diff
@@ -89,78 +120,168 @@ axs + scale_x_continuous(breaks = c(2000, 4000))
+There are also a few options for break functions in the scales package:
+
+* `scales::breaks_extended()`	creates automatic breaks for numeric axes
+* `scales::breaks_log()`	creates breaks appropriate for log axes
+* `scales::breaks_pretty()`	creates "pretty" breaks for date/times
+* `scales::breaks_width()`	creates equally spaced breaks
+
+See the documentation of the scales package for more detail.
+
+### Adjusting minor breaks
 
+You can adjust the minor breaks (the faint grid lines that appear between the major grid lines) by supplying a numeric vector of positions to the `minor_breaks` argument. This is particularly useful for log scales. First I'll create a vector of minor break values (on the transformed scale), using `%o%` to quickly generate a multiplication table and `as.numeric()` to flatten the table to a vector.
+\index{Minor breaks}\index{Log!ticks}
+```{r}
+mb <- as.numeric(1:10 %o% 10 ^ (0:4))
+```
+The following plots illustrate the effect of setting the minor breaks:
 `r columns(2, 2/3)`
 ```{r}
-df <- data.frame(x = c(2, 3, 5, 10, 200, 3000), y = 1)
-ggplot(df, aes(x, y)) + 
-  geom_point() + 
-  scale_x_log10()
+base <- ggplot(toy, aes(log, const)) + geom_point()
 
-mb <- as.numeric(1:10 %o% 10 ^ (0:4))
-ggplot(df, aes(x, y)) + 
-  geom_point() + 
-  scale_x_log10(minor_breaks = mb)
+base + scale_x_log10()
+base + scale_x_log10(minor_breaks = mb)
+```
+ 
+Note also that the scales package contains `minor_breaks_n()` and `minor_breaks_width()` functions that can be helpful in controlling the minor breaks.
+
+
+### Date scales {#date-scales}
+
+A special case of interest occurs when an aesthetic is mapped to date or date/time data. In ggplot2 these are treated as continuous variables that have special labels, and there are some helper functions designed to make it easier to specify the breaks and labels for these scales. The ggplot2 package supports the `Date` (for dates) and `POSIXct` (for date/times) classes, as well as the `hms` class provided by the hms package [@hms]. If your dates are in a different format you will need to convert them using `as.Date()`, `as.POSIXct()` or `hms::as_hms()`. You may also find the toolkit provided by the lubridate package helpful to manipulate time and date data [@lubridate].
+
+
+For appropriately formatted data mapped to the x aesthetic, ggplot2 will default to `scale_x_date()` and `scale_x_datetime()`, and there are similar scales for other aesthetics. In most respects date scales behave the same way as other continuous scales, but have some special behaviour that is worth noting. Specifically, in addition to the usual `breaks` and `labels` arguments, date scales have special `date_breaks` and `date_labels` arguments that allow you to work in date-friendly units:
```

Comment:
I can't see much in the documentation, but looking at the source it appears that `offset` is added after the `width` range is passed to `scales::fullseq()` so it needs to be something that can be added to a date object (numeric, difftime etc). I've mentioned this in the revised PR and given an example using a numeric offset. I'm not completely happy with it because the example feels longer than it needs to be, but I can't think of anything better right now